presto-docs/src/main/sphinx/functions/lambda.rst

```diff
@@ -24,6 +24,13 @@ Most SQL expressions can be used in a lambda body, with a few exceptions:
 Lambda Functions
 ----------------
 
+.. function:: apply(T, function<T,U>) -> U
+
+    Apply the lambda function on the given variable::
+
+        SELECT apply(5, x -> x + 1); -- 6
+        SELECT apply(1, x -> apply(x + 7, y -> y * 5) + 1); -- 41
```

Comment:
Let's use an example demonstrate the application of apply in using some input multiple times, e.g.

```
SELECT apply(5, IF(x % 2 == 0, x / 2, x * 3 + 1); -- 16
SELECT apply(16, IF(x % 2 == 0, x / 2, x * 3 + 1)); -- 8
```

Reply:
Isn't this more like an example of `IF`? :-|

Reply:
Something involving substr and length is a good example, where you need the value and the length of the value (for the offset or length calculation).